### DISCUSSION TITLE
Upgrade Transmission to the new 4.0.3 version

### DISCUSSION THREAD

**Discussion by ashr123** [Homebrew/discussions#4449] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.14-96-g302b224
ORIGIN: https://github.com/Homebrew/brew
HEAD: 302b224c2101a6b26cbb405f06e9a58c7b03f67d
Last commit: 2 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: db371ec7510651e00ebfb242887aa6119b22199f
Core tap last commit: 8 weeks ago
Core tap branch: master
Core tap JSON: 21 Apr 14:04 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.aMeWrWPGYs/org.xquartz:0
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.40.0 => /opt/homebrew/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 13.3.1-arm64
CLT: 14.3.0.0.1.1679647830
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  pycparser
  cffi
```

### Description of issue

The current version seems to be 3.00, but the latest version (as of now) is 4.0.3, in addition, when I searched for `transmission`, the following warning has been printed:

> Warning: Use transmission-cli instead of deprecated transmission

Is there a way to install the full app, including the GUI and web interface for example?

**Comment by ZhongRuoyu** [comment, marked as answer]
Based on your description, I'm assuming that you have installed the [`transmission-cli`](https://formulae.brew.sh/formula/transmission-cli#default) formula, rather than the [`transmission`](https://formulae.brew.sh/cask/transmission#default) cask.

Both `transmission-cli` formula and `transmission` cask have version 4.0.3. Have you tried `brew upgrade transmission-cli`?

> Warning: Use `transmission-cli` instead of deprecated `transmission`

That simply means `transmission` is renamed to `transmission-cli`.

> Is there a way to install the full app, including the GUI and web interface for example?

You probably want `brew install --cask transmission`.

**Reply by ashr123** [reply]
This is the full output for the following command:
```bash
brew info transmission
```

> Warning: Use transmission-cli instead of deprecated transmission
> Warning: Treating transmission as a formula. For the cask, use homebrew/cask/transmission
> ==> transmission-cli: stable 3.00 (bottled)
> Lightweight BitTorrent client
> https://www.transmissionbt.com/
> Not installed
> From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/transmission-cli.rb
> License: GPL-2.0-only or GPL-3.0-only
> ==> Dependencies
> Build: pkg-config ✔
> Required: libevent ✘, openssl@1.1 ✔
> ==> Caveats
> This formula only installs the command line utilities.
> 
> Transmission.app can be downloaded directly from the website:
>   https://www.transmissionbt.com/
> 
> Alternatively, install with Homebrew Cask:
>   brew install --cask transmission
> 
> To restart transmission-cli after an upgrade:
>   brew services restart transmission-cli
> Or, if you don't want/need a background service you can just run:
>   /opt/homebrew/opt/transmission-cli/bin/transmission-daemon --foreground --config-dir /opt/homebrew/var/transmission/ --log-info --logfile /opt/homebrew/var/transmission/transmission-daemon.log
> ==> Analytics
> install: 33 (30 days), 1,436 (90 days), 11,567 (365 days)
> install-on-request: 33 (30 days), 1,436 (90 days), 11,565 (365 days)
> build-error: 0 (30 days)

That means the the current version is 3.0.0...

About the full app installetion, the confusion is probably caused because of my lack of knowledge between formula and cask.

When should I use `--cask` and when not?

**Reply by ashr123** [reply]
In addition, I've executed the following command:
```sh
brew info transmission-cli
```
And indeed got

> ==> transmission-cli: stable 4.0.3 (bottled)
> Lightweight BitTorrent client
> https://www.transmissionbt.com/
> Not installed
> From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/transmission-cli.rb
> License: GPL-2.0-only or GPL-3.0-only
> ==> Dependencies
> Build: cmake ✘, gettext ✔, pkg-config ✔
> Required: libevent ✘, openssl@1.1 ✔
> ==> Caveats
> This formula only installs the command line utilities.
> 
> Transmission.app can be downloaded directly from the website:
>   https://www.transmissionbt.com/
> 
> Alternatively, install with Homebrew Cask:
>   brew install --cask transmission
> 
> To restart transmission-cli after an upgrade:
>   brew services restart transmission-cli
> Or, if you don't want/need a background service you can just run:
>   /opt/homebrew/opt/transmission-cli/bin/transmission-daemon --foreground --config-dir /opt/homebrew/var/transmission/ --log-info --logfile /opt/homebrew/var/transmission/transmission-daemon.log
> ==> Analytics
> install: 33 (30 days), 1,436 (90 days), 11,567 (365 days)
> install-on-request: 33 (30 days), 1,436 (90 days), 11,565 (365 days)
> build-error: 0 (30 days)

i.e. version 4.0.3

So there are 2 questions:

1. When should I use `--cask`?
2. If I'll execute
   ```bash
   brew install transmission
   ```
   What would have happend? would it install version 3.00 or version 4.0.0?

**Reply by ZhongRuoyu** [reply]
That is very strange. What if you try `brew update` before `brew info transmission`; do you still get 4.0.3? If so, can you share the output of `brew info --debug transmission`?

1. > When should I use `--cask`?

   Since you mentioned you want to "install the full app, including the GUI and web interface", you should install the cask, as mentioned in the caveats in your `brew info` output.
   Casks are "macOS applications distributed as binaries". If you are looking for an app that shows up in your Launchpad, then usually you will want to install casks.
2. > If I'll execute
   > ```
   > brew install transmission
   > ```
   > What would have happend? would it install version 3.00 or version 4.0.0?

   You should get version 4.0.3.

**Reply by ashr123** [reply]
> That is very strange. What if you try `brew update` before `brew info transmission`; do you still get 4.0.3? If so, can you share the output of `brew info --debug transmission`?

```bash
brew update -f
```

> ==> Updated Homebrew from 302b224c2 to 4ddebd9bd.
> Updated 2 taps (homebrew/core and homebrew/cask).
> ==> New Formulae
> ares                                       millet                                     poke                                       vulkan-tools
> ==> Outdated Formulae
> ...
> 
> You have 4 outdated formulae installed.
> You can upgrade them with brew upgrade
> or list them with brew outdated.

```bash
brew info --debug transmission
```

> Warning: Use transmission-cli instead of deprecated transmission
> /opt/homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /opt/homebrew/Library/Taps/homebrew/homebrew-core/Formula/transmission-cli.rb
> /opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading transmission
> Warning: Treating transmission as a formula. For the cask, use homebrew/cask/transmission
> ==> transmission-cli: stable 3.00 (bottled)
> Lightweight BitTorrent client
> https://www.transmissionbt.com/
> Not installed
> From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/transmission-cli.rb
> License: GPL-2.0-only or GPL-3.0-only
> ==> Dependencies
> /opt/homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading pkg-config from API
> Build: pkg-config ✔
> /opt/homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading libevent from API
> /opt/homebrew/Library/Homebrew/brew.rb (Formulary::FormulaAPILoader): loading openssl@1.1 from API
> Required: libevent ✘, openssl@1.1 ✔
> ==> Caveats
> This formula only installs the command line utilities.
> 
> Transmission.app can be downloaded directly from the website:
>   https://www.transmissionbt.com/
> 
> Alternatively, install with Homebrew Cask:
>   brew install --cask transmission
> 
> To restart transmission-cli after an upgrade:
>   brew services restart transmission-cli
> Or, if you don't want/need a background service you can just run:
>   /opt/homebrew/opt/transmission-cli/bin/transmission-daemon --foreground --config-dir /opt/homebrew/var/transmission/ --log-info --logfile /opt/homebrew/var/transmission/transmission-daemon.log
> /usr/bin/env /opt/homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/4.0.14-115-g4ddebd9\ \(Macintosh\;\ arm64\ Mac\ OS\ X\ 13.3.1\)\ curl/7.87.0 --header Accept-Language:\ en --retry 3 --fail https://formulae.brew.sh/api/formula/transmission-cli.json
> ==> Analytics
> install: 30 (30 days), 1,399 (90 days), 11,525 (365 days)
> install-on-request: 30 (30 days), 1,398 (90 days), 11,523 (365 days)
> build-error: 0 (30 days)

**Reply by ZhongRuoyu** [reply]
OK, so the problem is that `brew info transmission` is reading the formula from your local tap. Can you [open a `Homebrew/brew` issue](https://github.com/Homebrew/brew/issues/new?assignees=&labels=bug&template=bug.yml)? Please add a link to this discussion thread in the issue.

For now, to install 4.0.3, please use the formula's canonical name `transmission-cli`. (Or use the cask.)

**Reply by ashr123** [reply]
See https://github.com/Homebrew/brew/issues/15299

**Reply by ZhongRuoyu** [reply]
Thanks @ashr123.
